src/libstd/sys/sgx/abi/usercalls/mod.rs

```diff
@@ -149,7 +151,22 @@ pub fn exit(panic: bool) -> ! {
 
 /// Usercall `wait`. See the ABI documentation for more information.
 #[unstable(feature = "sgx_platform", issue = "56975")]
-pub fn wait(event_mask: u64, timeout: u64) -> IoResult<u64> {
+pub fn wait(event_mask: u64, mut timeout: u64) -> IoResult<u64> {
+    if timeout != WAIT_NO && timeout != WAIT_INDEFINITE {
+        // We don't want people to rely on accuracy of timeouts to make
+        // security decisions in an SGX enclave. That's why we add a random
+        // amount not exceeding +/- 10% to the timeout value to discourage
+        // people from relying on accuracy of timeouts while providing a way
+        // to make things work in other cases. Note that in the SGX threat
+        // model the enclave runner which is serving the wait usercall is not
+        // trusted to ensure accurate timeouts.
+        if let Ok(timeout_signed) = i64::try_from(timeout) {
+            let tenth = 1 + timeout_signed / 10;
+            let deviation = (rdrand64() as i64).checked_rem(tenth).unwrap_or(0);
+            timeout = timeout_signed.saturating_add(deviation) as _;
+        }
+        timeout = cmp::min(u64::MAX - 1, cmp::max(1, timeout));
```

Comment:
Why is this still here?

Reply:
to make sure we don't change the semantics requested by the caller, i.e. if `timeout != WAIT_NO && timeout != WAIT_INDEFINITE` it should stay that way after we fudge the value.

Reply:
It will never be WAIT_INDEFINITE because that doesn't fit in an i64. It would never be WAIT_NO if you didn't do `1 +` above. Even so, is there a fundamental difference between `wait(WAIT_NO)` and `wait(1ns)` considering CPU speed?

Reply:
fair enough, I'll remove the check